query/call_iterator.go

```diff
@@ -1247,6 +1247,214 @@ func newMovingAverageIterator(input Iterator, n int, opt IteratorOptions) (Itera
+			fn := NewTripleExponentialAverageReducer(n, nHold, warmupType)
+			return fn, fn
+		}
+		return newIntegerStreamFloatIterator(input, createFn, opt), nil
+	case UnsignedIterator:
+		createFn := func() (UnsignedPointAggregator, FloatPointEmitter) {
+			fn := NewTripleExponentialAverageReducer(n, nHold, warmupType)
+			return fn, fn
+		}
+		return newUnsignedStreamFloatIterator(input, createFn, opt), nil
+	default:
+		return nil, fmt.Errorf("unsupported triple exponential average iterator type: %T", input)
+	}
+}
+
+// newKaufmansEfficiencyRatioIterator returns an iterator for operating on a kaufmans_efficiency_ratio() call.
+func newKaufmansEfficiencyRatioIterator(input Iterator, n int, nHold int, opt IteratorOptions) (Iterator, error) {
+	switch input := input.(type) {
+	case FloatIterator:
+		createFn := func() (FloatPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansEfficiencyRatioReducer(n, nHold)
+			return fn, fn
+		}
+		return newFloatStreamFloatIterator(input, createFn, opt), nil
+	case IntegerIterator:
+		createFn := func() (IntegerPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansEfficiencyRatioReducer(n, nHold)
+			return fn, fn
+		}
+		return newIntegerStreamFloatIterator(input, createFn, opt), nil
+	case UnsignedIterator:
+		createFn := func() (UnsignedPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansEfficiencyRatioReducer(n, nHold)
+			return fn, fn
+		}
+		return newUnsignedStreamFloatIterator(input, createFn, opt), nil
+	default:
+		return nil, fmt.Errorf("unsupported kaufmans efficiency ratio iterator type: %T", input)
+	}
+}
+
+// newKaufmansAdaptiveMovingAverageIterator returns an iterator for operating on a kaufmans_adaptive_moving_average() call.
+func newKaufmansAdaptiveMovingAverageIterator(input Iterator, n int, nHold int, opt IteratorOptions) (Iterator, error) {
+	switch input := input.(type) {
+	case FloatIterator:
+		createFn := func() (FloatPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansAdaptiveMovingAverageReducer(n, nHold)
+			return fn, fn
+		}
+		return newFloatStreamFloatIterator(input, createFn, opt), nil
+	case IntegerIterator:
+		createFn := func() (IntegerPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansAdaptiveMovingAverageReducer(n, nHold)
+			return fn, fn
+		}
+		return newIntegerStreamFloatIterator(input, createFn, opt), nil
+	case UnsignedIterator:
+		createFn := func() (UnsignedPointAggregator, FloatPointEmitter) {
+			fn := NewKaufmansAdaptiveMovingAverageReducer(n, nHold)
+			return fn, fn
+		}
+		return newUnsignedStreamFloatIterator(input, createFn, opt), nil
+	default:
+		return nil, fmt.Errorf("unsupported kaufmans adaptive moving average iterator type: %T", input)
+	}
+}
+
+// newChandeMomentumOscillatorIterator returns an iterator for operating on a triple_exponential_moving_average() call.
+func newChandeMomentumOscillatorIterator(input Iterator, n int, nHold int, warmupType string, opt IteratorOptions) (Iterator, error) {
```

Comment:
I would like it if warmup type was passed in as the type rather than a string and it was parsed before this, but I also don't care enough to block the PR since I can change this myself. I'm just making this comment as a note that I'll probably change the argument for this at some point.